### PR TITLE
Changed save format to json. It will still load old save files.

### PR DESCRIPTION
**Pull request by dulsi**
Although files created before the slug field was added to the object will fail to save afterward.

This fixes issue #163 . I have not split monster instances and the species. This will allow techniques to be updated for monsters if the data changes but things like hit points and defenses will not.

### CODE REVIEW

File: tuxemon/core/components/save.py [hunk 1]
```diff
@@ -123,42 +129,96 @@ def load(slot):
...
+            tempstorage = dict()
+            for keys, values in json_data['storage'].items():
+                if keys == 'items':
+                    tempinv = dict()
+
+                    for slug,quant in values.items():
+                        tempinv1 = dict()
+                        tempinv1['item'] = Item(slug)
+                        tempinv1['quantity'] = quant
+                        tempinv[tempinv1['item'].name] = tempinv1
```
Comment (bitcraft) [inline tuxemon/core/components/save.py]: The name attribute is dependent on the locale set.  Is there a way to avoid its use here?

**Comment by bitcraft** [comment]
Thanks for tackling this issue.  JSON is ok with me, though you are building dictionaries so pickle/shelve could be used as well if others think the pickle binary format is preferred.  If we keep the JSON format, then I don't think we need code to load the old files....i'd rather it not be around to reduce code maintenance.

I'd also prefer to move the "load/save_from_savefile" from the fusion and monster files to the save/load modules.  If the file format changes in the future, it may not be obvious to look in the monster and fusion modules.

Looks good so far!

**Comment by dulsi** [comment]
Pickle/shelve is not a good idea. [Here](https://lwn.net/Articles/595352/) is an article detailing why.

I can certainly move the load/save functions from fusion/monster files. Is this just a personal preference or common practice under python?  The counter argument to your format change is that if new fields are added to monster and fusion, it may not be obvious that they need to look at save/load code. I'll make the change.

**Comment by bitcraft** [comment]
To be clear, I'm not advocating pickle.  I brought it up because I wasn't sure if we all agreed it should be JSON...which I'm fine with.   Anyway, thanks for understanding about moving the save/load code; I don't think it is a 'python' thing, mostly just a nice way to separate concerns.  Since the save/load code is dependent on the monster class, but the monster class is not dependent on the save/load, then the glue code should be organized with the save/load code.  Hope that helps.

**Comment by dulsi** [comment]
All the changes you asked for are in. Let me know if you see anything else you feel needs to be fixed up.

**Comment by bitcraft** [comment]
Works great!  I've opened #178, which is somewhat related...how do we handle save files that the game cannot read?  (I realize you had a work around for loading old-style saves, but I don't think we need that function...)  Maybe we can get a dialog or something to handle it.

I've tested the changes and it seems to work fine; saving and loading a file worked as expected.  Thanks again!

**Comment by ShadowApex** [comment]
> I brought it up because I wasn't sure if we all agreed it should be JSON...which I'm fine with.

I'm totally happy with saves being in JSON.

I also tested this out and confirmed it works great. Thanks for the PR!

The only reason I originally made saves using `pickle` was to have a small level of obfuscation for save files to discourage cheating by modifying your save file with a simple text editor. 

To help prevent this, maybe we could use a combination of json + base64 + zlib to keep save files small and not as easy to modify. This obviously won't stop people from cheating, but it may discourage casual modifying of save games. Something like this:

```python
import zlib
import base64
...
save_data = base64.b64encode(json.dump(json_data))
save_file.write(zlib.compress(save_data))
save_file.close()
```

**Comment by Sanglorian** [comment]
Isn't it kind of cool if people edit their own save files? I can imagine that being a kid's first interaction with programming and code and so on. Having the details of their game readable to the player sounds like a feature to me, not a bug!

**Pull request merged**